dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -922,4 +1016,113 @@ public String getURLFromEntry(String dockerEntry) {
+        }
+        dagJson.put("nodes", nodes);
+        dagJson.put("edges", edges);
+
+        return convertToJSONString(dagJson);
+    }
+
+    /**
+     * This method will setup the tools of CWL workflow
+     * It will then call another method to transform it through Gson to a Json string
+     * @param toolID toolDocker
+     * @return String
+     * */
+    private String getJSONTableToolContentCWL(Map<String, Pair<String, String>> toolID, Map<String, Pair<String, String>> toolDocker) {
+        // set up JSON for Table Tool Content CWL
+        ArrayList<Object> tools = new ArrayList<>();
+
+        //iterate through each step within workflow file
+        for(String key : toolID.keySet()){
+            //get the idName and fileName
+            String toolName = toolID.get(key).getLeft();
+            String fileName = toolID.get(key).getRight();
+
+            //get the docker requirement
+            String dockerPullName = toolDocker.get(key).getLeft();
+            String dockerLink = toolDocker.get(key).getRight();
+
+            //put everything into a map, then ArrayList
+            Map<String, String> dataToolEntry = new LinkedHashMap<>();
+            dataToolEntry.put("id", toolName);
+            dataToolEntry.put("file", fileName);
+            dataToolEntry.put("docker", dockerPullName);
+            dataToolEntry.put("link",dockerLink);
+            tools.add(dataToolEntry);
+        }
+
+        //call the gson to string transformer
+        return convertToJSONString(tools);
+    }
+
+    /**
+     * This method will setup the tools of WDL workflow
+     * It will then call another method to transform it through Gson to a Json string
+     * @param taskContent
+     * @return String
+     * */
+    private String getJSONTableToolContentWDL(Map<String, Pair<String, String>> taskContent){
+        // set up JSON for Table Task Content WDL
+        ArrayList<Object> tasks = new ArrayList<>();
+
+        //iterate through each task within workflow file
+        for(String key : taskContent.keySet()){
+            String dockerPull = taskContent.get(key).getLeft();
+            String dockerLink = taskContent.get(key).getRight();
+
+            //put everything into a map, then ArrayList
+            Map<String, String> dataTaskEntry = new LinkedHashMap<>();
+            dataTaskEntry.put("id", key);
+            dataTaskEntry.put("docker", dockerPull);
+            dataTaskEntry.put("link", dockerLink);
+            tasks.add(dataTaskEntry);
+        }
+
+        //call the gson to string transformer
+        return convertToJSONString(tasks);
+    }
+
+    /**
+     * This method will transform object containing the tools/dag of a workflow to Json string
+     * @param content
+     * @return String
+     * */
+    private String convertToJSONString(Object content){
+        //create json string and return
+        Gson gson = new Gson();
+        String json = gson.toJson(content);
+        System.out.println(json);
```

Comment:
Don't leave `System.out` calls in the code after you're done. Other use `LOG.debug' or delete them.